### PR TITLE
Update Transcript models for automatic language detection and fix demo app UI rendering of Live Transcription Config for mobile browsers

### PR DESCRIPTION
**Pull request by akashuc**
**Issue #:** Chime-46836

**Description of changes:**

* Additional fix for PR: https://github.com/aws/amazon-chime-sdk-js/pull/2025
* Fix meetingV2 Live Transcription Configuration UI to work in mobile browsers for serverless endpoint
* Rename `languageIdentifications` to `languageIdentification` in order to match Transcribe naming and media updates
* Mark `languageIdentification` as optional
* Update test cases

**Testing**
* Verified the changes are working by running local instance on the browser and taking in server-side changes, containing  updated protobuf of the models, transcription feature work as expected

*Can these tested using a demo application? Please provide reproducible step-by-step instructions.*

Yes, demo application for serverless-endpoint and demo/browser work as expected

Regression test can be performed on the changes:

* Use meetingV2 demo app
* Start and join the meeting
* Enable Live Transcription with new parameters or already released paramters
* Transcript should be enabled or both engine-transcribe and medical.

**Checklist:**

Have you successfully run npm run build:release locally?
Yes

Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?
No

Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?
No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/transcript/TranscriptEvent.ts [hunk 1]
```diff
@@ -69,21 +69,27 @@ export class TranscriptEventConverter {
...
-          for (const languageIdentification of result.languageIdentifications) {
+          if (result.languageIdentification && result.languageIdentification.length > 0) {
+            transcriptResult.languageIdentification = [];
+          }
+
+          if (result.languageCode) {
+            transcriptResult.languageCode = result.languageCode;
+          }
+
+          for (const languageIdentification of result.languageIdentification) {
```
Comment (noahmako) [inline src/transcript/TranscriptEvent.ts:86]: is there any problem if `result.languageIdentification` is undefined?

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,18 @@ All notable changes to this project will be documented in this file.
...
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
+
+### Added
+
+### Removed
+
+### Changed
+- Changed TrancriptResult field name `languageIdentifications` to `languageIdentification` and mark `languageIdentification` as optional
```
Comment (noahmako) [inline CHANGELOG.md:15]: I don't think we want these changelog entries for the finalized changelog

Comment (Chitz) [reply CHANGELOG.md]: Added it to trigger github workflow. Yes, ideally, we would only need for single changelog entry for feature release.

**Review by noahmako — APPROVED** [review]

**Review by ltrung — DISMISSED** [review]

File: src/transcript/TranscriptEvent.ts (hunk 1 above)
Comment (Chitz) [reply src/transcript/TranscriptEvent.ts]: Fixed.

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,16 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
```
Comment (ltrung) [inline CHANGELOG.md:8]: Can you remove this? We no longer do this anymore. You can instead update the original change log to have more details.

Comment (Chitz) [reply CHANGELOG.md]: Updated changelog. This is changelog is now consistent with release changelog in other SDKs.

**Review by noahmako — APPROVED** [review]

**Review by ltrung — APPROVED** [review]

**Pull request merged**